src/components/cl/basic/cl_basic_coll.c

```diff
@@ -6,11 +6,52 @@
 
 #include "cl_basic.h"
 
+static inline int ucc_is_inplace(ucc_coll_args_t *args) {
+    return (args->mask & UCC_COLL_ARGS_FIELD_FLAGS) &&
+           (args->flags & UCC_COLL_ARGS_FLAG_IN_PLACE);
+}
+
+ucc_status_t ucc_cl_basic_select_tl(ucc_base_coll_args_t *args,
+                                    ucc_cl_basic_team_t *cl_basic_team,
+                                    ucc_tl_team_t **tl_team)
+{
+    /* If NCCL TL is available and both src and dst buffers are CUDA then
+       choose NCCL TL for alltoall and alltoallv otherwise use UCP TL*/
+    if (cl_basic_team->tl_nccl_team != NULL) {
+        switch (args->args.coll_type) {
+        case UCC_COLL_TYPE_ALLTOALL:
+            if ((args->args.src.info.mem_type == UCC_MEMORY_TYPE_CUDA) &&
```

Comment:
inplace means src is ignored according to spec args table, right? is this condition correct?
suppose: inplace_is set, dst mem type is HOST, and src mem_type has garbage (since it is not required for inplace), then condition depends on the uninitialized src.info.mem_type.

Reply:
you are right, need to swap src and dst